op-node/rollup/derive/system_config.go

```diff
@@ -36,18 +36,38 @@ var (
 // UpdateSystemConfigWithL1Receipts filters all L1 receipts to find config updates and applies the config updates to the given sysCfg
 func UpdateSystemConfigWithL1Receipts(sysCfg *eth.SystemConfig, receipts []*types.Receipt, cfg *rollup.Config, l1Time uint64) error {
 	var result error
+
+	// copy sysConfig to an update accumulator
+	// to allow for pre-parsing of all updates prior to application
+	updated := eth.SystemConfig{
+		BatcherAddr:          sysCfg.BatcherAddr,
+		Overhead:             sysCfg.Overhead,
+		Scalar:               sysCfg.Scalar,
+		GasLimit:             sysCfg.GasLimit,
+		EIP1559Params:        sysCfg.EIP1559Params,
+		OperatorFeeParams:    sysCfg.OperatorFeeParams,
+		MinBaseFee:           sysCfg.MinBaseFee,
+		DAFootprintGasScalar: sysCfg.DAFootprintGasScalar,
+	}
```

Comment:
It would be very easy to miss adding a new field to this struct. Can we create a copy with:

```suggestion
	updated := *sysCfg
```

Reply:
Ha yes can do. I had it like this originally, but wanted to have the copied bits be explicit. But you're right that this handles it more cleanly, *unless* one of the internal objects is non-primitive.